include/tvm/ffi/reflection/registry.h

```diff
@@ -35,13 +38,37 @@ namespace tvm {
 namespace ffi {
 /*! \brief Reflection namespace */
 namespace reflection {
-
 /*!
  * \brief Trait that can be used to set field info
  * \sa DefaultValue, AttachFieldFlag
  */
 struct FieldInfoTrait {};
 
+/*! \brief User-supplied metadata attached to a field or a method */
+class Metadata : public FieldInfoTrait {
```

Comment:
ideally we want to keep reflection not depend on extra, i know it can be a bit painful, but hopefully we can string stitching here

Reply:
Updated with a significantly more complicated approach:
- Added `EscapeString` in `string.h`, which is refactored from `WriteString` in `json_write.cc`
- Updated `refl::Metadata::ToJSON` to using string stitching to dump a Python. It also on-the-fly checks if value types is one of `int | bool | str` and disallow other types.